acinclude/krb5.m4

```diff
@@ -8,125 +8,156 @@
+    AC_RUN_IFELSE([
+      AC_LANG_PROGRAM([[#include "compat/krb5.h"]],[[krb5_context c; krb5_init_context(&c);]])
+    ],[squid_cv_working_krb5=yes],[squid_cv_working_krb5=no],[:])])
+  SQUID_STATE_ROLLBACK(squid_krb5_test)
+  AS_IF([test "x$squid_cv_working_krb5" = "xyes"],[
+    AC_DEFINE(HAVE_KRB5,1,[KRB5 support])
+  ],[test "x$squid_cv_working_krb5" = "xno" -a `echo $LIBS | grep -i -c "(-)L"` -gt 0],[
+    AC_MSG_NOTICE([Check Runtime library path !])
+  ])
+])
 
-        return 0;
-}
-]])], [ squid_cv_broken_heimdal_krb5_h=yes ], [ squid_cv_broken_heimdal_krb5_h=no ])
+dnl check how to access the max skew in the krb5 context
+AC_DEFUN([SQUID_CHECK_KRB5_GET_MAX_TIME_SKEW],[
+  AC_CACHE_CHECK([for max_skew in struct krb5_context],squid_cv_max_skew_context,[
+    SQUID_STATE_SAVE(squid_krb5_test)
+    CPPFLAGS="-I${srcdir:-.} $CPPFLAGS"
+    AC_CHECK_LIB(krb5,[krb5_get_max_time_skew],[
+      squid_cv_max_skew_context="yes"],[
+      AC_COMPILE_IFELSE([
+        AC_LANG_PROGRAM([[#include "compat/krb5.h"]],[[krb5_context kc; kc->max_skew = 1;]])
+      ],[squid_cv_max_skew_context=replace],[squid_cv_max_skew_context=no],[squid_cv_max_skew_context=no])
     ])
-    SQUID_STATE_ROLLBACK(squid_krb5_heimdal_test)
+    SQUID_STATE_ROLLBACK(squid_krb5_test)
+  ])
+  AS_IF([test "x$squid_cv_max_skew_context" = "xreplace"],[
+    AC_DEFINE(HAVE_MAX_SKEW_IN_KRB5_CONTEXT,1,[Define to 1 if krb5_get_max_time_skew() needs replacing])
+  ])
+  AS_IF([test "x$squid_cv_max_skew_context" != "xno"],[
+    AC_DEFINE(HAVE_KRB5_GET_MAX_TIME_SKEW,1,[Define to 1 if you have krb5_get_max_time_skew])
   ])
-]) dnl SQUID_CHECK_KRB5_HEIMDAL_BROKEN_KRB5_H
+])
 
-dnl check the max skew in the krb5 context, and sets squid_cv_max_skew_context
-AC_DEFUN([SQUID_CHECK_MAX_SKEW_IN_KRB5_CONTEXT],[
-  AC_CACHE_CHECK([for max_skew in struct krb5_context],
-                  squid_cv_max_skew_context, [
+dnl check whether krb5_get_init_creds_opt_free() requires a context
+AC_DEFUN([SQUID_CHECK_KRB5_GET_INIT_CREDS_FREE_CONTEXT],[
+  AC_CACHE_CHECK([for krb5_get_init_creds_free requires krb5_context],
+    squid_cv_krb5_get_init_creds_free_context,[
     SQUID_STATE_SAVE(squid_krb5_test)
     CPPFLAGS="-I${srcdir:-.} $CPPFLAGS"
-    AC_COMPILE_IFELSE([
-      AC_LANG_PROGRAM([[
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "compat/krb5.h"
-krb5_context kc; kc->max_skew = 1;
-      ]])
-    ],[ squid_cv_max_skew_context=yes ],
-    [ squid_cv_max_skew_context=no ])
+      ]],[[krb5_context context;
+        krb5_get_init_creds_opt *options = nullptr;
+        krb5_get_init_creds_opt_free(context, options)]])],
+    [squid_cv_krb5_get_init_creds_free_context=yes],
+    [squid_cv_krb5_get_init_creds_free_context=no],[:])
     SQUID_STATE_ROLLBACK(squid_krb5_test)
   ])
+  AS_IF([test "x$squid_cv_krb5_get_init_creds_free_context" = "xyes"],[
+    AC_DEFINE(HAVE_KRB5_GET_INIT_CREDS_FREE_CONTEXT,1,[Define to 1 if you krb5_get_init_creds_free requires krb5_context])
```

Comment:
```suggestion
    AC_DEFINE(HAVE_KRB5_GET_INIT_CREDS_FREE_CONTEXT,1,[Define to 1 if krb5_get_init_creds_opt_free() requires krb5_context])
```

* "you" is a typo; probably meant to be "your", but the word is not really needed there
* There is no krb5_get_init_creds_free(). 

This is old moved code, so I do not insist on this fix. Your call.